contracts/tgrade-valset/src/contract.rs

```diff
@@ -367,6 +370,28 @@ fn execute_slash(
     Ok(resp)
 }
 
+fn execute_simulate_validators(
+    deps: DepsMut,
+    _info: MessageInfo,
+    validators: Vec<ValidatorInfo>,
+) -> Result<Response, ContractError> {
+    // Assert admin is not found (contract not instantiated!) for this to be possible
+    let admin_res = ADMIN.get(deps.as_ref());
+    if let Err(err) = admin_res {
+        let not_found = matches!(err, StdError::NotFound { .. });
+        if !not_found {
+            return Err(ContractError::AdminError(AdminError::NotAdmin {}));
+        }
+    } else {
+        return Err(ContractError::AdminError(AdminError::NotAdmin {}));
+    }
```

Comment:
There's probably a simpler / nicer way to do this.

Reply:
we should add such method to admin.

Reply:
wait... if this is only for tests, we do not want this in production build at all.
how about feature-gating it with `#[cfg(features = "integration")]` and you make such a build for the wasm used in integration test? Then no worries about it as this code path isn't in production at all

Reply:
Good idea. Will do this tomorrow.

Reply:
Forgot to remove this in the first iteration.

Then, tried to do it in a follow-up, but hit dependency / feature problems with compiling the wasm using the integration feature.

I'll post my working copy and we can see if there's a way around those problems (other than keeping / removing the entry point entirely).